memory/stressng_mem.py

```diff
@@ -0,0 +1,193 @@
+            self.log.info(
+                "=====================================================")
+            if (return_code == 137):
+                self.log.info("== > stress-ng memory test timed out and \
+                              was forcefully terminated!")
+            else:
+                self.log.info(
+                    "==> Error %s reported on stressor %s!",
+                    return_code, stressor)
+            self.log.info(
+                "=====================================================")
+
+        return return_code
+
+    def calculate_variable_time(self):
+        self.total_memory_in_GiB = memory.meminfo.MemTotal.g
+        self.log.info("Total Memory on the system : %s GiB",
+                      self.total_memory_in_GiB)
+
+        extra_time = (self.time_per_gig * self.total_memory_in_GiB)
+        self.variable_time = (self.base_time + extra_time)
+        self.log.info("Time limit set for constant_run_time stressors is %s "
+                      "seconds per stressor.", self.base_time)
+        self.log.info("Time limit set for variable_run_time stressors is %s "
+                      "seconds per stressor.", self.variable_time)
+
+    def setUp(self):
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+
+        self.had_error = 0
+        self.base_time = self.params.get("base_time", default=300)
+        self.time_per_gig = self.params.get("time_per_gig", default=10)
+        self.url = self.params.get("url", default="https://github.com/"
+                                   "ColinIanKing/stress-ng/archive/master.zip")
+
+        deps = ['gcc', 'make']
+
+        deps.extend(['libattr-devel', 'libcap-devel', 'libgcrypt-devel',
+                     'zlib-devel', 'libaio-devel'])
+
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel("%s is needed for the test to be run." % package)
+
+        tarball = self.fetch_asset(
+            'stressng.zip', locations=self.url, expire='7d')
+        archive.extract(tarball, self.workdir)
+        sourcedir = os.path.join(self.workdir, 'stress-ng-master')
+        os.chdir(sourcedir)
+
+        result = build.run_make(sourcedir, process_kwargs={
+                                'ignore_status': True})
+
+        for line in str(result).splitlines():
+            if 'error:' in line:
+                self.cancel(
+                    "Build Failed, Please check the build logs for details !!")
+
+        build.make(sourcedir, extra_args='install')
+        clear_dmesg()
+
+    def test_memory(self):
+        self.swap_memory = memory.meminfo.SwapTotal.m
+
+        if (self.swap_memory == 0):
+            self.cancel("Swap space unavailable! Please activate swap space "
+                        "and re-run this test!")
+
+        self.calculate_variable_time()
+
+        crt_stressors = ["bsearch", "context", "hsearch", "lsearch", "matrix",
```

Comment:
can we drive this yaml as well

Reply:
> can we drive this yaml as well

@PraveenPenguin 

Yes, this could be driven through the yaml. 

But here are a few concerns with this approach - 
- if we decide to do that then, how should we provide the default value? 
- Since we cannot default the entire list of stressors. it would become mandatory for the user to use yaml file everytime the program is run. (w/o the stressors the code is of no use)

Reply:
@rohan-r-deshpande you can use the default values in script itself or mux in yaml if test is running on each varient

Reply:
@Naresh-ibm how can I default the entire list of stressors in the script?

Reply:
Done.